satpy/readers/fci_l1c_nc.py

```diff
@@ -211,15 +214,44 @@ def __init__(self, filename, filename_info, filetype_info):
 
         self._cache = {}
 
+    @property
+    def nominal_start_time(self):
+        """Get nominal start time."""
+        RC_date = self.filename_info['start_time'].replace(hour=0, minute=0, second=0, microsecond=0)
+        if self.filename_info['coverage'] == 'FD':
+            RC_period_min = 10
+        else:
+            RC_period_min = 2.5
+        return RC_date + timedelta(minutes=(self.filename_info['repeat_cycle_in_day']-1)*RC_period_min)
+
+    @property
+    def nominal_end_time(self):
+        """Get nominal end time."""
+        if self.filename_info['coverage'] == 'FD':
+            RC_period_min = 10
+        else:
+            RC_period_min = 2.5
```

Comment:
Make `RC_period_min` a class variable defined according to the `coverage` already in the `init`? That way you avoid code duplication.

Reply:
resolved by https://github.com/pytroll/satpy/pull/2420/commits/d75d3539d3910399782d033c6e73d4f28d32df7b